internal/metrics/otelmetrics/factory_test.go

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package otelmetrics_test
+
+import (
+	"testing"
+	"time"
+
+	promReg "github.com/prometheus/client_golang/prometheus"
+	promModel "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+
+	"github.com/jaegertracing/jaeger/internal/metrics/otelmetrics"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"github.com/jaegertracing/jaeger/pkg/testutils"
+)
+
+func TestMain(m *testing.M) {
+	testutils.VerifyGoLeaks(m)
+}
+
+func newTestFactory(t *testing.T, registry *promReg.Registry) metrics.Factory {
+	exporter, err := prometheus.New(prometheus.WithRegisterer(registry), prometheus.WithoutScopeInfo())
+	require.NoError(t, err)
+	meterProvider := sdkmetric.NewMeterProvider(sdkmetric.WithReader(exporter))
+	return otelmetrics.NewFactory(meterProvider)
+}
+
+func findMetric(t *testing.T, registry *promReg.Registry, name string) *promModel.MetricFamily {
+	metricFamilies, err := registry.Gather()
+	require.NoError(t, err)
+
+	for _, mf := range metricFamilies {
+		t.Log(mf.GetName())
+		if mf.GetName() == name {
+			return mf
+		}
+	}
+	require.Fail(t, "Expected to find Metric Family")
+	return nil
+}
+
+func promLabelsToMap(labels []*promModel.LabelPair) map[string]string {
+	labelMap := make(map[string]string)
+	for _, label := range labels {
+		labelMap[label.GetName()] = label.GetValue()
+	}
+	return labelMap
+}
+
+func TestInvalidCounter(t *testing.T) {
+	registry := promReg.NewPedanticRegistry()
+	factory := newTestFactory(t, registry)
```

Comment:
nit: presumably you always call these two together, so can collapse into one statement
```suggestion
	factory, registry := newTestFactory(t)
```